### PR TITLE
Feature 1701 deferred formatting cleanup

### PR DESCRIPTION
**Pull request by artntek**
Changes that were deferred until after some branches were merged back to develop, in order to avoid merge conflict hell. See Issue #1701

### CODE REVIEW

File: src/edu/ucsb/nceas/metacat/DocumentImpl.java [hunk 1]
```diff
@@ -2555,60 +2406,58 @@ private static XMLReader initializeParser(DBConnection dbconn,
...
-                    && needValidation) {
+            } else if (ruleBase != null && ruleBase.equals(DTD) && needValidation) {
                 logMetacat.info("DocumentImpl.initalizeParser - Using dtd parser");
                 // turn on dtd validaton feature
                 parser.setFeature(VALIDATIONFEATURE, true);
-                eresolver = new DBEntityResolver(dbconn,
-                        (DBSAXHandler) chandler, dtd);
+                eresolver = new DBEntityResolver(dbconn, chandler, dtd);
                 dtdhandler = new DBDTDHandler(dbconn);
-                parser.setEntityResolver((EntityResolver) eresolver);
```
Comment (taojing2002) [inline src/edu/ucsb/nceas/metacat/DocumentImpl.java:2601]: Did test without casting?

File: src/edu/ucsb/nceas/metacat/DocumentImpl.java [hunk 2]
```diff
@@ -2555,60 +2406,58 @@ private static XMLReader initializeParser(DBConnection dbconn,
...
-                parser.setDTDHandler((DTDHandler) dtdhandler);
```
Comment (taojing2002) [inline src/edu/ucsb/nceas/metacat/DocumentImpl.java:2602]: Did test without casting?

File: src/edu/ucsb/nceas/metacat/DocumentImpl.java (hunk 1 above)
Comment (artntek) [reply src/edu/ucsb/nceas/metacat/DocumentImpl.java:2601]: `eresolver` is already of type `EntityResolver` - [see definition above](https://github.com/NCEAS/metacat/pull/1725/files#diff-bdee27d50eaffc92fec27fd02305f2554e85018ed989baa75b3bd3c608c34747R2383):

```java
            EntityResolver eresolver;
```
...so casting is unnecessary

File: src/edu/ucsb/nceas/metacat/DocumentImpl.java (hunk 2 above)
Comment (artntek) [reply src/edu/ucsb/nceas/metacat/DocumentImpl.java:2602]: `dtdhandler` is already of type `DTDHandler` - [see definition above](https://github.com/NCEAS/metacat/pull/1725/files#diff-bdee27d50eaffc92fec27fd02305f2554e85018ed989baa75b3bd3c608c34747R2384):

```java
            DTDHandler dtdhandler;
```
...so casting is unnecessary

File: src/edu/ucsb/nceas/metacat/dataone/CNodeService.java [hunk 3]
```diff
@@ -87,2025 +55,2135 @@
...
+            }
+            // insert the system metadata into the object store
+            logMetacat.debug("Starting to insert SystemMetadata...");
+            try {
+                //for the object whose authoriative mn is v1. we need reset the modification date.
+                //d1-sync already set the serial version. so we don't need do again.
+                D1NodeVersionChecker checker =
+                    new D1NodeVersionChecker(sysmeta.getAuthoritativeMemberNode());
+                String version = checker.getVersion("MNRead");
+                boolean changeModificationDate =
```
Comment (taojing2002) [inline src/edu/ucsb/nceas/metacat/dataone/CNodeService.java:1300]: Did you do the change or the editor does?

File: src/edu/ucsb/nceas/metacat/dataone/CNodeService.java [hunk 4]
```diff
@@ -87,2025 +55,2135 @@
...
+                            + pid.getValue());
+                    sysmeta.setSerialVersion(BigInteger.ONE);
+                    //for the object whose authoritative mn is v1. we need reset the modification
+                    // date.
+                    //for the object whose authoritative mn is v2. we just accept the
+                    // modification date.
+                    D1NodeVersionChecker checker =
+                        new D1NodeVersionChecker(sysmeta.getAuthoritativeMemberNode());
+                    String version = checker.getVersion("MNRead");
+                    boolean changeModificationDate =
```
Comment (taojing2002) [inline src/edu/ucsb/nceas/metacat/dataone/CNodeService.java:1716]: Did you make the change or the editor make?

File: src/edu/ucsb/nceas/metacat/dataone/CNodeService.java (hunk 3 above)
Comment (artntek) [reply src/edu/ucsb/nceas/metacat/dataone/CNodeService.java:1300]: disregard - will revert changes and recommit without code cleanup

File: src/edu/ucsb/nceas/metacat/dataone/CNodeService.java (hunk 4 above)
Comment (artntek) [reply src/edu/ucsb/nceas/metacat/dataone/CNodeService.java:1716]: disregard - will revert changes and recommit without code cleanup

File: src/edu/ucsb/nceas/metacat/DocumentImpl.java (hunk 2 above)
Comment (artntek) [reply src/edu/ucsb/nceas/metacat/DocumentImpl.java:2602]: disregard - will revert changes and recommit without code cleanup

File: src/edu/ucsb/nceas/metacat/DocumentImpl.java (hunk 1 above)
Comment (artntek) [reply src/edu/ucsb/nceas/metacat/DocumentImpl.java:2601]: disregard - will revert changes and recommit without code cleanup

File: src/edu/ucsb/nceas/metacat/MetacatHandler.java [hunk 5]
```diff
@@ -345,448 +320,463 @@ public static InputStream read(String docid, String dataType) throws ParseLSIDEx
...
+                    String filename = filepath + docid;
+                    inputStream = readFromFilesystem(filename);
+                } catch (PropertyNotFoundException pnf) {
+                    logMetacat.debug("There was a problem finding the "
+                                         + "application.datafilepath property. The error "
+                                         + "message was: " + pnf.getMessage());
+                    throw pnf;
+                } // end try()
+            } else {
+                // this is n metadata document
```
Comment (taojing2002) [inline src/edu/ucsb/nceas/metacat/MetacatHandler.java:346]: Should it be an or n?

Comment (mbjones) [reply src/edu/ucsb/nceas/metacat/MetacatHandler.java:346]: It should be "a metadata document". The difference between `a` and `an` is about the pronunciation of the following word -- typically use `a` unless the following word starts with a vowel sound, in which case use `an` (e.g., "an action", "an egg", "an hour",  but "a house"). See: https://english.stackexchange.com/a/155

**Review by taojing2002 — APPROVED** [review]

**Pull request merged**